### PR TITLE
Allow changing lid settings without restarting the system

### PR DESCRIPTION
**Pull request by GranPC**

### CODE REVIEW

**Comment by GranPC** [comment]
I originally left it in just in case we ever add something that needs the infobox again, but I'll get rid of it!

File: cli/main.vala [hunk 1]
```diff
@@ -17,11 +17,18 @@
  * Boston, MA  02110-1301, USA.
  */
 
+[DBus (name = "org.freedesktop.systemd1.Manager")]
+interface SystemDBus : Object {
+    public abstract GLib.ObjectPath reload_or_try_restart_unit (string unit, string mode) throws IOError;
```
Comment (tintou) [inline cli/main.vala]:
```suggestion
    public abstract GLib.ObjectPath reload_or_try_restart_unit (string unit, string mode) throws GLib.Error;
```

File: cli/main.vala [hunk 2]
```diff
@@ -17,11 +17,18 @@
...
+}
+
 public class LoginDHelper.Application : GLib.Application {
     private const uint ACTIVE_TIMEOUT_SECONDS = 5;
     private uint timeout_id = 0;
     private uint own_id = -1;
 
+    private static SystemDBus? systemd_bus_proxy = null;
```
Comment (tintou) [inline cli/main.vala]:
```suggestion
    private SystemDBus? systemd_bus_proxy = null;
```
no need for static here, LoginDHelper is already sure to be unique per process

File: cli/main.vala [hunk 3]
```diff
@@ -69,6 +85,8 @@ public class LoginDHelper.Application : GLib.Application {
             Bus.unown_name (own_id);
         }
 
+        reload_logind ();
```
Comment (tintou) [inline cli/main.vala]: There is no need for a dedicated function here, as the function only has 4 lines, but adding a comment with something like "We need to restart systemd-logind to ensure that the lid settings are taken into account" might help to understand why this is required

**Comment by tintou** [comment]
Sorry for the late review btw

**Comment by GranPC** [comment]
No problem! I'll get back to you with the changes you requested soon. :)

**Comment by danrabbit** [comment]
@GranPC Can you follow up here please? :)

**Review by tintou — APPROVED** [review]

**Pull request merged**